### PR TITLE
[DOC] Documentation for builtin KMeans function

### PR DESCRIPTION
**Pull request by t201**
[KMeans.txt](https://github.com/apache/systemml/files/4709296/KMeans.txt)

Edit (by @j143):
This function adds documentation for the `KMeans` builtin function.

### CODE REVIEW

**Comment by j143** [comment]
We need to add code like this:  https://github.com/apache/systemml/commit/8d020fb9f807750b07d08398caa2d433305819b6
:)

**Comment by j143** [comment]
@t201 - I will review this tonight. Thank you.

File: docs/builtins-reference.md [hunk 1]
```diff
@@ -105,9 +106,51 @@ print(toString(D))
...
+| Type           | Description |
+| :------------- | :---------- |
+| String         | The mapping of records to centroids |
+| String         |The output matrix with the centroids |
+
+
+### Example
+
+KMeans = rand(Matrix x, centers k=0,int iter.max=25,int nstart=10)
+return(Matrix[numerical] C,Matrix[numerical] Y)
```
[the remark this answers is not in the document]
Comment (j143) [reply docs/builtins-reference.md:149]: 2. Example in this case means `a runnable code snippet`.

**For Example:** In the case of `slicefinder` function
image

We can copy paste this usage code into the Colab and run it!

image

Which would give the following output:

image

**Comment by j143** [comment]
Hi @t201 - Thanks for working on this function. This is good starting point and there are some minor changes related to formatting and example.

File: docs/builtins-reference.md [hunk 2]
```diff
@@ -127,12 +123,12 @@ kmeans(x, centers, iter.max, nstart)
...
 ### Arguments
 
-| Name   | Type            | Default  | Description |
-| :----  | :-------------  | -------- | :------------------------------- |
-| x      | Matrix[Numeric] | required |The input Matrix to do KMeans on..|
-|centers | Int             | -------- |The no. of cenetrs                |
-|iter.max|Int              |10        |Max no. of iterations allowed     |
-|nstart  |Int              |10        |No. of random starting positions  |
+| Name     | Type            | Default  | Description |
+| :--      | :--             | --       | :--         |
```
Comment (j143) [inline docs/builtins-reference.md:127]: Can you revert changes of this line. I was only talking about `centers` corresponding!

File: docs/builtins-reference.md [hunk 3]
```diff
@@ -143,12 +139,13 @@ kmeans(x, centers, iter.max, nstart)
 | String         |The output matrix with the centroids |
 
 
+
+**DML-bodied built-in functions** are written as DML-Scripts and executed as such when called.
 ### Example
 
-KMeans = rand(Matrix x, centers k=0,int iter.max=25,int nstart=10)
-return(Matrix[numerical] C,Matrix[numerical] Y)
+```h2o.kmeans(x = predictors, k = 100, estimate_k = T, standardize = F,
```
Comment (j143) [inline docs/builtins-reference.md:146]: Are you able to run this code in the colab notebook provided to you, if yes screenshot please.

Also, this seems to be a copy-paste from `H2O` software, different from `SystemDS`.

**Comment by j143** [comment]
@nurattarun - Refer https://github.com/apache/systemml/pull/933#discussion_r434296676

File: docs/builtins-reference.md (hunk 2 above)
Comment (t201) [reply docs/builtins-reference.md]: Done. Only the changes to 'centers' were done this time.

File: docs/builtins-reference.md (hunk 3 above)
Comment (t201) [reply docs/builtins-reference.md]: I'm unable to find a small code snippet for Kmeans function in R which runs without any error on Collab. I tried with a lot of examples.

Comment (j143) [reply docs/builtins-reference.md]: We need to closely understand the this function:
https://github.com/apache/systemml/blob/faec15810e18fc497e950b830811a919f5aa5c5f/scripts/builtin/kmeans.dml#L46

Think about the inputs required and initialize the required arguments.
We can take already implemented functions as examples. 
This task is tough and iterative, copy-paste solutions are not available. Some people spend whole days to find the correct combinations, though. 🙂

Comment (t201) [reply docs/builtins-reference.md]: Alright . I'm on it.

Comment (t201) [reply docs/builtins-reference.md]: The syntax errors are quite persistent . No example worked in the collab till now.

Comment (j143) [reply docs/builtins-reference.md]:
```
X = rand (rows = 10, cols = 10)
kmeans(X = X, k = 0, runs = 10, max_iter = 1000, )
```

Comment (t201) [reply docs/builtins-reference.md]: Screenshot (144)

Comment (j143) [reply docs/builtins-reference.md]: The snippet, I've provided above is a DML code to run. 🙂

Comment (j143) [reply docs/builtins-reference.md]: Kindly have a look at how other functions are documented.

**Comment by j143** [comment]
Thank you, @t201 - LGTM. 👍 
🎉 🚀

**Pull request closed without merging**